### PR TITLE
feat: parquet files can be read form all storages we support

### LINKED ISSUE #1342: Support Querying parquet files that are not in local disks
__Use case:__
Our parquet files can be stored in Object Store today but the work in [#1082](https://app.zenhub.com/workspaces/iox--conductor-602ce49e1ed7db001da6bb4c/issues/influxdata/influxdb_iox/1082) that query the parquet files only works if the files are on local disk. This is a follow-up task of #108 to support all kind of storages

__Proposal:__
Extend the work in #1082 to read parquet files that are not in local disk. This includes data in memory and all object store we support today.

__Current behaviour:__
Only works if parquet files on local disks

__Desired behaviour:__
It should work with all kinds of storages we support

__Alternatives considered:__

### PR DESCRIPTION
**Pull request by NGA-TRAN**
Closes #1342 

This PR add support to read parquet files from non-local disk

Describe your proposed changes here.

- [ ] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: parquet_file/src/lib.rs [hunk 1]
```diff
@@ -12,7 +12,6 @@ pub mod chunk;
 pub mod metadata;
 pub mod storage;
 pub mod table;
+pub mod utils;
```
Comment (NGA-TRAN) [inline parquet_file/src/lib.rs:15]: Utils is now including useful functions imported from storage and db modules

File: parquet_file/src/storage.rs [hunk 2]
```diff
@@ -432,37 +316,34 @@ impl Storage {
...
-        // TODO: support non local file object store. Ticket #1342
-        let full_path = match (&store.0, path) {
-            (ObjectStoreIntegration::File(file_root), Path::File(location)) => {
-                file_root.path(&location)
-            }
-            (_, _) => {
-                panic!("Non local file object store not supported")
-            }
-        };
+        let parquet_data = futures::executor::block_on(async move {
```
Comment (NGA-TRAN) [inline parquet_file/src/storage.rs:319]: @tustvold and @carols10cents This is the line Carol says we may need to change to something else. The purpose of it is to invoke an async function from non-async one.

File: parquet_file/src/utils.rs [hunk 3]
```diff
@@ -21,28 +21,66 @@ use internal_types::{
...
 /// Load parquet from store and return table name and parquet bytes.
+// This function is for test only
 pub async fn load_parquet_from_store(chunk: &Chunk, store: Arc<ObjectStore>) -> (String, Vec<u8>) {
-    let table = chunk.table_names(None).next().unwrap();
-    let path = chunk.table_path(&table).unwrap();
-    let parquet_data = store
-        .get(&path)
+    load_parquet_from_store_for_chunk(chunk, store)
         .await
         .unwrap()
```
Comment (NGA-TRAN) [inline parquet_file/src/utils.rs:55]: This function is for test only but I think I will remove the `unwrap` here and return Result and then will add `unwrap` in the test functions that invoke it

File: server/src/db.rs [hunk 4]
```diff
@@ -1878,36 +1879,20 @@ mod tests {
...
-        // Note: We do not care about escaping quotes here because it is just a test
-        let sql = format!(
-            "CREATE EXTERNAL TABLE parquet_table STORED AS PARQUET LOCATION '{}'",
-            path.display()
-        );
-
-        let mut ctx = context::ExecutionContext::new();
-        let df = ctx.sql(&sql).unwrap();
-        df.collect().await.unwrap();
+        // Now read data from that path
```
Comment (NGA-TRAN) [inline server/src/db.rs:1885]: Use our own read parquet tests instead of the SQL one from data fusion

File: parquet_file/src/utils.rs (hunk 3 above)
Comment (alamb) [reply parquet_file/src/utils.rs:55]: I think it is a good idea

**Review by alamb — APPROVED** [review]
This PR looks like an improvement to me. Getting the parquet reader to be an async stream is probably important eventually, but I think it could be done as a follow on PR

File: parquet_file/src/storage.rs (hunk 2 above)
Comment (NGA-TRAN) [reply parquet_file/src/storage.rs:319]: Will file the ticket

File: parquet_file/src/utils.rs [hunk 5]
```diff
@@ -21,28 +21,66 @@ use internal_types::{
...
+}
+
+pub async fn load_parquet_from_store_for_chunk(
+    chunk: &Chunk,
+    store: Arc<ObjectStore>,
+) -> Result<(String, Vec<u8>)> {
+    let table = chunk
+        .table_names(None)
```
[the remark this answers is not in the document]
Comment (NGA-TRAN) [reply parquet_file/src/utils.rs:63]: Right. We should modify it

File: parquet_file/src/storage.rs (hunk 2 above)
Comment (alamb) [reply parquet_file/src/storage.rs:319]: https://github.com/influxdata/influxdb_iox/issues/1487 for anyone following along

**Comment by carols10cents** [comment]
I think this can be closed because the changes were merged in https://github.com/influxdata/influxdb_iox/pull/1486 instead, please reopen if I'm mistaken!

**Pull request closed without merging**